### PR TITLE
Ulp better next prev and epsilon

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/core/regular/prev.hpp [hunk 1]
```diff
@@ -15,16 +15,19 @@
 #include <eve/module/core/constant/minf.hpp>
 #include <eve/module/core/constant/one.hpp>
 #include <eve/module/core/regular/all.hpp>
+#include <eve/module/core/regular/fnma.hpp>
 #include <eve/module/core/regular/if_else.hpp>
 #include <eve/module/core/regular/is_gez.hpp>
 #include <eve/module/core/regular/is_positive.hpp>
 #include <eve/module/core/regular/is_negative.hpp>
+#include <eve/module/core/regular/is_normal.hpp>
```
Comment (jfalcou) [inline include/eve/module/core/regular/prev.hpp:23]: this is redundant

File: include/eve/module/core/regular/ulp.hpp [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+//!   {
+//!      // Regular overload
+//!      constexpr auto ulp(value auto x)                          noexcept; // 1
+//!
+//!      // Lanes masking
+//!      constexpr auto ulp[conditional_expr auto c](value auto x) noexcept; // 2
+//!      constexpr auto ulp[logical_value auto m](value auto x)    noexcept; // 2
+//!
+//!      // Semantic options
+//!      constexpr auto abs[downward](value auto x)                noexcept; // 1
```
Comment (jfalcou) [inline include/eve/module/core/regular/ulp.hpp]: ulp not abs

File: include/eve/module/core/regular/ulp.hpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+//!         between `x` and `next(x)`. (Kahan-Harrisson definitionn)
+//!      2. [The operation is performed conditionnaly](@ref conditional).
+//!      3  Computes [elementwise](@ref glossary_elementwise) the 'unit in the last place'
+//!         between `x` and `pred(x)`. (Goldberg definition)
+//!
+//!  @note the `upward` and `downward` only differ if x is a power of 2.
+//!
+//!  @groupheader{External references}
+//!   *  [wikipedia](//!https://en.wikipedia.org/wiki/Unit_in_the_last_place)
+//!   *  [Inria](https://hal.science/inria-00070503)
```
Comment (jfalcou) [inline include/eve/module/core/regular/ulp.hpp]: put the title of the paper. "Inria" is not very helpful

File: test/doc/core/ulp.cpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+int main()
+{
+  eve::wide wf0{eve::next(0.0), 1.0, 2.0, 3.0, -1.0, -2.0, -3.0, -4.0};
+  eve::wide wf1{eve::next(0.0f), 1.0f, 2.0f, 3.0f, -1.0f, -2.0f, -3.0f, -4.0f};
+
+  std::cout << "<- wf0                               = " << wf0 << "\n";
+  std::cout << "<- wf1                               = " << wf1 << "\n";
+
+//   std::cout << std::hexfloat << eve::exp2(-53.0)*(1+exp2(-52)) << std::endl;
```
Comment (jfalcou) [inline test/doc/core/ulp.cpp]: remove commented code

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/module/core/regular/prev.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/core/regular/prev.hpp:23]: done

File: include/eve/module/core/regular/ulp.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/core/regular/ulp.hpp]: the file has been deleted

**Review by jfalcou — APPROVED** [review]

**Pull request merged**